mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1004,6 +1015,15 @@ public DuplicateInsertPreventionStrategy getDuplicateInsertPreventionStrategy(En
 		return new MongoDBAggregationOutput( output, entityKeyMetadata );
 	}
 
+	private static final OptionsContext getTypeContext(TupleContext tupleContext) {
+		try {
+			return tupleContext.getTupleTypeContext().getOptionsContext();
+		}
+		catch (HibernateException exception) {
+			return null;
```

Comment:
What kind of exception occurs here that you can ignore?

Reply:
There could be OnlyWithTransactionContext which throws exception with message "The tuple context is not available, probably because we are dealing with more than a single entity type".